docs/hdl_coding_guideline.md

```diff
@@ -0,0 +1,710 @@
+statements, and end of line comments. The code **must** be written in a tabular 
+format.
+
+#### Example:
+
+##### Incorrect:
+```verilog
+    reg[3:0] my_signal1; //description
+reg[31:0] my_decoded_signal1; //description
+reg[4:0] my_signal2, my_signal3; //description
+wire[2:0] my_select; //description
+```
+
+##### Correct:
+```verilog
+reg  [ 3:0]  my_signal1;         //description
+reg  [31:0]  my_decoded_signal1; //description
+reg  [ 4:0]  my_signal2;         //description
+reg          my_signal3;         //description
+
+wire [ 2:0]  my_select;          //description
+```
+
+
+**A8**: Parentheses **must** be used around all boolean statements and in 
+complex equations, in order to force the order of operations and avoid confusion. 
+Complex boolean expressions *should* be expressed as multi-line aligned statements.
+
+#### Example:
+
+##### Incorrect:
+```verilog
+if ((my_signal1 && your_signal1) || (my_signal2 && your_signal2) || (my_signal3 && your_signal3)) begin
+  my_signal1 = TRUE;
+  my_delayed_signal1 = !your_signal;
+end
+```
+
+##### Correct:
+```verilog
+if ((my_signal1 && your_signal1) ||
+    (my_signal2 && your_signal2) ||
+    (my_signal3 && your_signal3)) begin
+  my_signal1 = TRUE;
+  my_delayed_signal1 = !your_signal;
+end
+```
+
+
+**A9**: A line **must** not contain more than one statement. **Do not** concatenate 
+multiple statements on the same line.
+
+#### Example:
+
+##### Incorrect:
+```verilog
+upper_en = (p5type && xadr1[0]); lower_en = (p5type && !xadr1[0]);
+```
+
+##### Correct:
+```verilog
+upper_en = (p5type && xadr1[0]);
+lower_en = (p5type && !xadr1[0]);
+```
+
+
+**A10**: In module instances:  
+
+**A10.1**: **All** parameters and ports, **must** be written on a separate line, 
+even if there are few of them or their names are short.
+#### Example:
+
+##### Incorrect:
+```verilog
+my_module #(.PARAMETER1 (PARAMETER1)) i_my_module (.clk (clk));
+```
+
+##### Correct:
+```verilog
+my_module #(
+  .PARAMETER1 (PARAMETER1)
+) i_my_module (
+  .clk (clk));
+```
+
+
+**A10.2**: When instantiating a module, the label of the module instance **must** be 
+on a separate line, with the closing parenthesis of the parameters list (if that's 
+the case) and the opening parenthesis of the ports list. The closing parenthesis of 
+the ports list must be right next to the last parenthesis of the last port.
+
+#### Example:
+```verilog
+my_module #(
+  .PARAMETER1 (PARAMETER1),
+  .PARAMETER2 (PARAMETER2)
+) i_my_module (
+  .clk (clk),
+  .rst (rst),
+  .data_in (data_in),
+  .en (en),
+  .response_out (response_out));
+```  
+
+**A10.3**: Commented parts of code **must** not be added to the master branch (i.e if, 
+case, module instances, etc).
+
+
+
+**A11**: In module declarations:  
+
+**A11.1**: Verilog modules **must** use Verilog-2001 style parameter
+declarations. This increases legibility and consistency.
+
+#### Example:
+```verilog
+module my_module #(
+  parameter PARAMETER1 = 0)(
```

Comment:
I would avoid putting the )( at the end of the line. 
A better example can be found here:
https://github.com/analogdevicesinc/hdl/blob/master/library/jesd204/jesd204_soft_pcs_rx/jesd204_soft_pcs_rx.v#L52

Reply:
Done